src/Protocol/Protocol17x.cpp

```diff
@@ -2145,7 +2145,7 @@ void cProtocol172::HandlePacketEntityAction(cByteBuffer & a_ByteBuffer)
 void cProtocol172::HandlePacketKeepAlive(cByteBuffer & a_ByteBuffer)
 {
 	HANDLE_READ(a_ByteBuffer, ReadBEInt32, Int32, KeepAliveID);
-	m_Client->HandleKeepAlive(KeepAliveID);
+	m_Client->HandleKeepAlive(static_cast<UInt32>(KeepAliveID));
```

Comment:
Change `KeepAliveID` to an `UInt32`, by changing the `HANDLE_READ`:
```cpp
HANDLE_READ(a_ByteBuffer, ReadBEUInt32, UInt32, KeepAliveID);
```
Then no cast will be needed.